sklearn/model_selection/tests/test_validation.py

```diff
@@ -776,6 +777,87 @@ def split(self, X, y=None, groups=None):
 
     assert_raises(ValueError, cross_val_predict, est, X, y, cv=BadCV())
 
+    X, y = load_iris(return_X_y=True)
+    assert_warns(RuntimeWarning, cross_val_predict, LogisticRegression(),
```

Comment:
Can you use assert_warns_message to test the warning message?

Reply:
Apart from this do you have any concerns with the PR?

Reply:
I don't think so, I am fine with the general strategy as I mentioned above. There are some minor things I will fix shortly.

If you haven't already, maybe you can have a quick look at the warnings/error message and make sure they are fine.

Reply:
Another thing: are you fine with float_min as the default value for missing classes for predict_log_proba and decision_function. Another option is `-inf` but maybe some consumers downstream will raise an error will infinite values.

Reply:
predict_log_proba should probably use -inf... decision_function is a messy case.

Reply:
I don't think float-min is perfect, but I'm documenting it at least.